docs/website/docs/reference/frequently-asked-questions.md

```diff
@@ -0,0 +1,114 @@
+
+In light of this limitation, `dlt` advises a workaround for incremental loading. This involves specifying both the start and end dates or page numbers for your data extraction process and conducting the backfill using smaller, manageable ranges, which are usually divided into around 100 loads. Implementing this strategy provides a higher degree of control and significantly mitigates the risk of losing any progress should the pipeline experience a failure or if it needs to be stopped midway through execution. For a more comprehensive understanding of how to implement this workaround, it is recommended to consult `dlt's` documentation on incremental loading with a cursor field, which can be found here: [Incremental Loading with a Cursor Field](../../docs/general-usage/incremental-loading#incremental-loading-with-a-cursor-field).
+
+## Is truncating or deleting a staging table safe?
+
+You can safely truncate those or even drop the whole staging dataset. However, it will have to be recreated on the next load and might incur extra loading time or cost.
+
+## How can I develop a "custom" pagination tracker?
+
+There are two ways. One, you can use `dlt.sources.incremental` to create a custom cursor for tracking pagination in data streams that lack an explicit cursor field.
+
+Second, while it's possible to utilize `dlt.sources.incremental` for tracking changes in data streams, you're not limited to this method for managing pagination or custom cursors. Instead, you can leverage the flexibility of managing state directly in Python. Access and modify the state like a standard Python dictionary: `state = dlt.current.resource_state(); state["your_custom_key"] = "your_value"`. This approach allows you to define your custom pagination logic based on your specific needs.
+
+However, be cautious about overusing the state dictionary, especially in cases involving substreams for each user, as it might become unwieldy. A better strategy might involve tracking users incrementally. Then, upon updates, you only refresh the affected users' substreams entirely. This consideration helps maintain efficiency and manageability in your custom pagination implementation.
+
+## What's the best way to add an "_orchestrator_pipeline_run_id" to datasets during ingestion to trace back to the specific run or code?
+
+To include an `_orchestrator_pipeline_run_id`, utilize environmental variables by adding this identifier to your data (e.g., in a dictionary or dataframe) before yielding it in your resource. Alternatively, create a dedicated resource for a lookup table mapping `_dlt_load_id` to `_orchestrator_pipeline_run_id`. This method ensures minimal storage impact while maintaining efficient traceability of your data assets.
```

Comment:
I don't get it, could you add example? which env variables?

Reply:
I added some more context, I couldn't find an example.